src/components/filters/attribute-filter.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  Select,
+  SelectOption,
+  SelectOptionObject,
+  SelectVariant,
+  TextInput,
+  ToolbarChip,
+  ToolbarChipGroup,
+  ToolbarFilter,
+  ToolbarGroup,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons';
+import React from 'react';
+import { useDebounce } from '../../hooks/useDebounce';
+import { TestIds } from '../../test-ids';
+import './attribute-filter.css';
+import { Attribute, AttributeList, Filters } from './filter.types';
+import { SearchSelect } from './search-select';
+
+interface AttributeFilterProps {
+  attributeList: AttributeList;
+  filters?: Filters;
+  onFiltersChange?: (filters: Filters) => void;
+  isDisabled?: boolean;
+}
+
+export const AttributeFilter: React.FC<AttributeFilterProps> = ({
+  attributeList,
+  filters = {},
+  onFiltersChange,
+  isDisabled,
+}) => {
+  const [isAttributeExpanded, setIsAttributeExpanded] = React.useState(false);
+  const changeFromInput = React.useRef(false);
+  const [textInputValue, setTextInputValue] = React.useState('');
+  const debouncedInputValue = useDebounce(textInputValue);
+  const [selectedAttributeId, setSelectedAttributeId] = React.useState<
+    string | undefined
+  >(attributeList[0]?.id);
+
+  const handleAttributeToggle = () => {
+    setIsAttributeExpanded(!isAttributeExpanded);
+  };
+
+  const textAttribute = attributeList.find(
+    (attr) => attr.id === selectedAttributeId && attr.valueType === 'text',
+  );
+
+  React.useEffect(() => {
+    if (textAttribute) {
+      const [initialText] = filters[textAttribute.id] ?? [];
+      changeFromInput.current = false;
+      setTextInputValue(initialText ?? '');
+    }
+  }, [textAttribute, filters]);
+
+  const handleAttributeSelect = (
+    _: React.MouseEvent | React.ChangeEvent,
+    value: string | SelectOptionObject,
+  ) => {
+    if (typeof value === 'string') {
+      setSelectedAttributeId(value);
+    }
+    setIsAttributeExpanded(false);
+  };
+
+  const handleAttributeValueChange = (value: Set<string>) => {
+    if (selectedAttributeId) {
+      if (!filters[selectedAttributeId]) {
+        filters[selectedAttributeId] = new Set();
+      }
+
+      onFiltersChange?.({
+        ...filters,
+        [selectedAttributeId]: value,
+      });
+    }
+  };
+
+  React.useEffect(() => {
+    const [initialText] = textAttribute ? filters[textAttribute.id] ?? [] : [];
+    const filterReset = initialText === undefined && debouncedInputValue === '';
+
+    if (
+      debouncedInputValue !== undefined &&
+      !filterReset &&
+      changeFromInput.current
+    ) {
+      handleAttributeValueChange(
+        debouncedInputValue.length === 0
```

Comment:
```suggestion
        debouncedInputValue === ''
```
Nit: To make it more obvious that `debouncedInputValue` is a string, not an array